### PR TITLE
Update Diagnostics to ASP.NET Core 2.1

### PR DESCRIPTION
**Pull request by henkmollema**
@jskeet these are the 2.1 changes requested here https://github.com/googleapis/google-cloud-dotnet/pull/3215#discussion_r302497546.

Updates the ASP.NET Core runtime packages to the latest 2.1 LTS release as well as some other package updates.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj [hunk 1]
```diff
@@ -22,18 +22,13 @@
...
+    <PackageReference Include="Microsoft.Extensions.Http" Version="2.1.1" />
     <PackageReference Include="Moq" Version="4.12.0" />
     <PackageReference Include="xunit" Version="2.4.1" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.4.1" />
     <PackageReference Include="Xunit.SkippableFact" Version="1.3.12" />
     <Analyzer Condition="Exists('..\..\..\tools\Google.Cloud.Tools.Analyzers\bin\$(Configuration)\netstandard1.3\publish\Google.Cloud.Tools.Analyzers.dll')" Include="..\..\..\tools\Google.Cloud.Tools.Analyzers\bin\$(Configuration)\netstandard1.3\publish\Google.Cloud.Tools.Analyzers.dll" />
-    <Reference Condition="'$(TargetFramework)' == 'net452'" Include="Microsoft.CSharp" />
-    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
-  </ItemGroup>
-  <ItemGroup>
```
Comment (henkmollema) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj]: I merged the `<ItemGroup>` elements, but it seems that the API checker only checks the first `<ItemGroup>` element which results in errors. Was it intentionally in a separate element?

Comment (jskeet) [reply apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj]: Yes, the intention is that the first ItemGroup is automatically populated based on apis.json; anything after that doesn't get touched.
Unfortunately that probably means messing with quite a few of your project file changes. Are you happy to do that, or would you like me to create a new commit on your branch?

Comment (jskeet) [reply apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj]: (I'll wait until that's done before reviewing, as I suspect the changes will be smaller. This is *much* appreciated though - thanks!)

Comment (henkmollema) [reply apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj]: Do you want to keep these dependencies separated or should they move to the same item group element and should apis.json be updated?

**Comment by jskeet** [comment]
Right, #3265 is merged now. Apologies for the inconvenience of all of this - especially as this PR is effectively a branch from the 3.0 work! I do think it'll make the history easier to follow though.
Let me know if you run into problems rebasing - I'm happy to do that and force push to your branch if you'd prefer.

**Comment by henkmollema** [comment]
Rebased. `generateprojects.sh` keeps adding this to the test project files:

https://github.com/googleapis/google-cloud-dotnet/blob/02097865ada5cc0e2be3e5c83f03a8eca074de3a/apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests.csproj#L33-L34

Do you have any clue why? And should we get rid of it? (how?)

**Comment by jskeet** [comment]
That's fine - it's not really needed here, but it's harmless. See https://github.com/googleapis/google-cloud-dotnet/blob/master/tools/Google.Cloud.Tools.ProjectGenerator/Program.cs#L490

**Comment by jskeet** [comment]
(I've added a merge commit to fix the rebase conflict, which was due to me fixing a stray comma elsewhere.)

This is a much simpler PR now :) Will review in the morning as I have meetings for the rest of the day.
Thanks again!

**Comment by henkmollema** [comment]
Thanks for merging! 😄

**Pull request merged**